app.rb

```diff
@@ -25,37 +26,49 @@ def initialize
     @bucket = @s3.bucket(settings.aws_bucket)
   end
 
-  get '/' do
-    if authed?
-      erb :index, locals: { projects: key_list }
-    else
-      redirect '/auth/myusa'
-    end
-  end
-
   get '/auth/myusa/callback' do
     halt(401, 'Not Authorized') unless env['omniauth.auth']
     session[:user_email] = env['omniauth.auth'].info.email
     redirect '/'
   end
 
+  before %r{^(?!\/auth)} do
+    redirect '/auth/myusa' unless authed?
+  end
+
+  get '/' do
+    erb :index, locals: { projects: key_list }
+  end
+
   get '/results' do
-    if authed?
-      erb :index, locals: { projects: key_list }
-    else
-      redirect '/auth/myusa'
-    end
+    erb :index, locals: { projects: key_list }
   end
 
   get '/results/:name' do |name|
-    halt(401, 'Not Authorized') unless authed?
     versions = get_version_list name
     'Invalid Project' if versions.count == 0
     erb :results, locals: { versions: versions }
   end
 
+  get '/results/:name/current' do |name|
+    file_data = get_file(name, nil)
+    if file_data.nil?
+      'Invalid Version'
+    else
+      erb :report, locals: { report_data: create_report(file_data) }
+    end
```

Comment:
To avoid the double-negative, I'd change this to

```ruby
if file_data
  # ...
else
  # ...
end
```

I know that some people prefer the "guard" style of having the success scenario at the end, so feel free to ignore if you like it better this way.